kubeflow/Makefile

```diff
@@ -160,6 +160,9 @@ hydrate-kubeflow:
 	kustomize build --load_restrictor none -o $(BUILD_DIR)/metacontroller $(KF_DIR)/metacontroller
 	mkdir -p $(BUILD_DIR)/kubeflow-issuer
 	kustomize build --load_restrictor none -o $(BUILD_DIR)/kubeflow-issuer $(KF_DIR)/kubeflow-issuer
+	mkdir -p $(BUILD_DIR)/cluster-local-gateway
+	kustomize build --load_restrictor none -o $(BUILD_DIR)/cluster-local-gateway $(KF_DIR)/cluster-local-gateway
```

Comment:
@jal06 I was suggesting changing this line to
```
kustomize build --load_restrictor none -o $(BUILD_DIR)/cluster-local-gateway $(MANIFESTS_DIR)/path/to/cluster-local-gateway
```
because there's no need to add user customizations to that package.

Reply:
@Bobgy, 

I didn't test it, but I am not sure it could work because `make apply` starts by cleaning `.build` so cluster-local-gateway would be cleaned before being applied.

```
.PHONY: apply
apply: clean-build check-name check-iap apply-gcp wait-gcp create-ctxt apply-asm apply-kubeflow iap-secret
```
...
```
clean-build:
	# Delete build because we want to prune any resources which are no longer defined in the manifests
	rm -rf $(BUILD_DIR)/
	mkdir -p $(BUILD_DIR)/
```


Moreover, IMHO, when putting cluster-local-gateway in instance, the process of cluster-local-gateway is consistent with the process of other applications (kubeflow-issuer, metacontroller, kubeflow, ...) so it could be easier to maintain.
WDYT ?

Reply:
@Bobgy, I am sorry, I missed th commit of cluster-local-gateway/kustomization.taml. I think it is OK now

Reply:
yes, consistency is a good point. Let me think about it more.

I was previously suggesting directly refer to upstream instead, because everything in `instance` folder needs user's manual maintenance (especially when upgrading), so if something doesn't need any customization, I preferred not putting it in instance folder.

Reply:
> if something doesn't need any customization, I preferred not putting it in instance folder.

Agree; that makes sense
But IMHO deleting build because we want to prune any resources which are no longer defined in the manifests makes also  sense and I don't see any easy solution to do it. 

> Let me think about it more.

OK; thanks

Reply:
@jal06
I guess you did not understand how this works
```
kustomize build --load_restrictor none -o $(BUILD_DIR)/cluster-local-gateway $(MANIFESTS_DIR)/path/to/cluster-local-gateway
```
Note that the `-o $(BUILD_DIR)/cluster-local-gateway` part is the output, and the `$(MANIFESTS_DIR)/path/to/cluster-local-gateway` part is the input.

I suggested changing `$(MANIFESTS_DIR)/path/to/cluster-local-gateway` to `....upstream/path/to/cluster-local-gateway`, the command will hydrate it everytime you apply to build folder. So that essentially means changing source of the hydration from instance to upstream, but build folder will always be hydrated each time. There won't be a problem there.